docs/CompVis_Stable_Diffusion_Instructions.md

```diff
@@ -0,0 +1,77 @@
+
+## Installing Dependency Packages
+
+We need a few Python packages, namely the Hugging Face script libraries for transformers and diffusers along with ONNX Runtime for DirectML.
+
+```
+pip install diffusers transformers onnxruntime-directml onnx
+```
+
+### Hardware Requirement
+Since it is still necessary for the entire model to fit within the GPU memory while executing. An ideal GPU to run this model with should have 8GB of VRAM available. Here are a few examples:
+- NVIDIA GeForce RTX 2070 or later
+- AMD Radeon RX 6500 XT (8GB) or later
+- Intel Arc A750 Graphics or later 
+
+## Downloading the Model
+
+We first need to download the model from Hugging Face. You need an account at Hugging Face, so if you haven't done so, now is the time. Once you've set up a Hugging Face account, go generate an access token (just follow their instruction in the web site).
+
+Once you have the account and an access token, authenticate yourself in a terminal or powershell console by running the following command.
+
+```
+huggingface-cli.exe login
+```
+
+It'll ask for your access token, which you can find on your account profile `Settings -> Access Tokens`, just copy it from here and carefully paste it on this prompt. Note that you won't see anything appear on the prompt when you paste it, that's fine. It's there already, just hit Enter. You'll start downloading the model from Hugging Face.
+
+## Converting to ONNX
+
+The model is trained with PyTorch so it can naturally convert to ONNX. Since we'll be using DirectML through ONNX Runtime, this step is needed. The script `convert_stable_diffusion_checkpoint_to_onnx.py` you already have here is just a local copy of the same file from the [Hugging Face diffusers GitHub repo](https://github.com/HuggingFace/diffusers/blob/main/scripts/convert_stable_diffusion_checkpoint_to_onnx.py), so you don't need to clone that repo.
+
+```
+python convert_stable_diffusion_checkpoint_to_onnx.py --model_path="CompVis/stable-diffusion-v1-4" --output_path="./stable_diffusion_onnx" --fp16
+```
+
+This will run the conversion and put the result ONNX files under the `stable_diffusion_onnx` folder. For better performance, we recommend you convert the model to half-precision floating point data type using the `--fp16` option.
+
+_Note: As of this writing, you cannot run this `--fp16` option until you have installed CUDA support with Torch packages using the following instruction. This will require up to 3 GB of additional disk space._
+
+```
+pip install torch>=1.13.0+cu116 torchvision>=0.13.0+cu116 torchaudio>=0.13.0 --extra-index-url https://download.pytorch.org/whl/cu116
+```
+
+## Running the ONNX Model
+
+You'll need a script that looks like what in the `text2image.py` file as follow. On a mid-range NVIDIA RTX 2070, a single image currently takes 20 seconds to generate from a prompt. It'll take up to 10 mins on the CPU.
```

Comment:
I'm curious how many steps you run with by default, 25 or 50? (not asking for any document change here, just curious, because mine take about 4 minutes on CPU, but that's 25 steps)

Reply:
I don't recall. It's approaching 10 mins on my machine before I decided to stop.